lib/whitehall/asset_manager_storage.rb

```diff
@@ -1,20 +1,35 @@
 class Whitehall::AssetManagerStorage < CarrierWave::Storage::Abstract
   def store!(file)
     original_file = file.to_file
-    temporary_location = File.join(
+    temporary_location = ::File.join(
       Whitehall.asset_manager_tmp_dir,
       SecureRandom.uuid,
-      File.basename(original_file)
+      ::File.basename(original_file)
     )
 
-    FileUtils.mkdir_p(File.dirname(temporary_location))
+    FileUtils.mkdir_p(::File.dirname(temporary_location))
     FileUtils.cp(original_file, temporary_location)
-    legacy_url_path = File.join('/government/uploads', uploader.store_path)
+    legacy_url_path = ::File.join('/government/uploads', uploader.store_path)
     AssetManagerCreateWhitehallAssetWorker.perform_async(temporary_location, legacy_url_path)
     file
   end
 
-  def retrieve!(_identifier)
-    raise "We're not currently reading assets from Asset Manager so this shouldn't be called."
+  def retrieve!(identifier)
+    asset_path = uploader.store_path(identifier)
+    File.new(asset_path)
+  end
+
+  class File
+    def initialize(asset_path)
+      @legacy_url_path = '/government/uploads/' + asset_path
```

Comment:
It might be better to use `File.join` here

Reply:
Done!